doc/how_to/custom_components/esm/panes.md

```diff
@@ -0,0 +1,178 @@
+    object = param.Dict()
+
+    _esm = """
+import { Chart } from "https://esm.sh/chart.js/auto"
+
+let chart = null;
+
+function createChart(canvasEl, model) {
+    if (chart) {
+        chart.destroy();
+    }
+    chart = new Chart(canvasEl.getContext('2d'), model.object);
+}
+
+export function render({ model }) {
+    const canvasEl = document.createElement('canvas');
+    // The chart will not render before after the layout has been created
+    requestAnimationFrame(() => createChart(canvasEl, model));
+
+    const updateChart = () => createChart(canvasEl, model);
+    model.on('object', updateChart);
+
+    return canvasEl;
+}
+"""
+
+
+def data(chart_type="line"):
+    return {
+        "type": chart_type,
+        "data": {
+            "labels": ["January", "February", "March", "April", "May", "June", "July"],
+            "datasets": [
+                {
+                    "label": "Data",
+                    "backgroundColor": "rgb(255, 99, 132)",
+                    "borderColor": "rgb(255, 99, 132)",
+                    "data": [0, 10, 5, 2, 20, 30, 45],
+                }
+            ],
+        },
+        "options": {
+            "responsive": True,
+            "maintainAspectRatio": False,
+        },
+    }
+
+
+chart_type = pn.widgets.RadioBoxGroup(
+    name="Chart Type", options=["bar", "line"], inline=True
+)
+chart = ChartJSComponent(
+    object=pn.bind(data, chart_type), height=400, sizing_mode="stretch_width"
+)
+pn.Column(chart_type, chart).servable()
+```
+
+Note the use of the `requestAnimationFrame` to postpone the rendering of the chart. If not used then Chart.js will not render. Dealing with layout issues like this sometimes requires a bit of iteration. If you get stuck, share your question and minimum, reproducible code example on [Discourse](https://discourse.holoviz.org/).
+
+## Creating a Cytoscape Pane
+
+This example will show you how to build a more advanced [CytoscapeJS](https://js.cytoscape.org/) pane.
+
+```{pyodide}
+import param
+import panel as pn
+
+from panel.custom import JSComponent
+
+
+class Cytoscape(JSComponent):
+
+    object = param.List()
+
+    layout = param.Selector(
+        default="cose",
+        objects=[
+            "breadthfirst",
+            "circle",
+            "concentric",
+            "cose",
+            "grid",
+            "preset",
+            "random",
+        ],
+    )
+    style = param.String("", doc="Use to set the styles of the nodes/edges")
+
+    zoom = param.Number(1, bounds=(1, 100))
+    pan = param.Dict({"x": 0, "y": 0})
+
+    data = param.List(doc="Use to send node's data/attributes to Cytoscape")
+
+    selected_nodes = param.List()
+    selected_edges = param.List()
+
+    _esm = """
+import { default as cytoscape} from "https://esm.sh/cytoscape"
+let cy = null;
+
+export function render({ model, el }) {
+    if (cy) { cy.destroy() }
+
+    cy = cytoscape({
+        container: el,
+        layout: {name: model.layout},
```

Comment:
We should get the warning `A Cytoscape container has style position:static and so can not use UI extensions properly`  fixed

![image](https://github.com/user-attachments/assets/281f2d71-fe21-4353-8141-14ef6f931637)

Reply:
Fixed. by creating a div with position relative. But then the div needs to be created `after_render` which makes the whole example a bit more convoluted.